include/nbl/builtin/hlsl/blit/formats_encode.hlsl

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_BLIT_FORMATS_ENCODE_INCLUDED_
+#define _NBL_BUILTIN_HLSL_BLIT_FORMATS_ENCODE_INCLUDED_
+
+#include <nbl/builtin/hlsl/cpp_compat/cpp_compat.hlsl>
+#include <nbl/builtin/hlsl/ieee754.hlsl>
+// NOTE: Assumes E_FORMAT is available in <nbl/builtin/hlsl/EFormat.hlsl> even tho it's not there yet.
+#include <nbl/builtin/hlsl/Eformat.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace blit
+{
+
+template <E_FORMAT format>
+struct encode_output { using type = float4; };
+template<> struct encode_output<EF_R32G32_SFLOAT> { using type = uint32_t4; };
+template<> struct encode_output<EF_B10G11R11_UFLOAT_PACK32> { using type = uint32_t4; };
```

Comment:
I think this entire header and `encode` functions can be nuked, because both formats you have special paths for here MUST be supported as storage images if `shaderStorageImageExtendedFormats` feature is enabled (which I'll soon require and you can force-enable)

Reply:
https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#formats-mandatory-features-64bit